### ISSUE TITLE
toml.Primitive should encode correctly

### ISSUE THREAD

**Issue by chowey** [BurntSushi/toml#63]
If I have a struct like this:

``` go
type SomeStruct struct {
    PreserveData toml.Primitive
    SomeData *MyData
}
```

I would expect that I can read a toml file, decode it to a `SomeStruct`, change the `SomeData`, and encode it back into a file with the `PreserveData` basically staying the same. Instead, it encodes to nothing.

Can we have a `toml.Primitive` encode back into whatever it used to be? Sort of like how `json.RawMessage` works?

**Comment by cespare** [comment]
If we do json-style custom encoders (`toml.Marshaler`, basically -- see #76) then this can cleanly be implemented by adding the appropriate methods to `toml.Primitive`.

Otherwise we have to special-case it in the encoder (as it's already special-cased in the decoder).

**Comment by arp242** [comment]
> If we do json-style custom encoders (toml.Marshaler, basically -- see #76) then this can cleanly be implemented by adding the appropriate methods to toml.Primitive.
>
> Otherwise we have to special-case it in the encoder (as it's already special-cased in the decoder)

Considering this hasn't been fixed almost 7 years after this PR was first opened and that the special case is just three lines of code, fixing it like this now sound reasonable. It solves actual issues people are facing and can always improve later on when that gets implemented.

I merged this via #281, which adds a comment for the above so it's less likely to get forgotten and improves the test that was added for this.
